src/path.util.ts

```diff
@@ -9,14 +9,16 @@ export const getValuePropertyPair = (instance: any, path: string):
       validPathSegments
           .slice(0, validPathSegments.length - 1)
           .reduce((curInstance, pathSegment) => {
-              if (!curInstance.hasOwnProperty(pathSegment)) {
-                  curInstance[pathSegment] = {};
+            const decodedSegment = decodeURIComponent(pathSegment);
```

Comment:
I don't think so, but it seems clearer to me since we only want decoded segments in the end anyway.